internal/tools/aws/database.go

```diff
@@ -184,22 +185,35 @@ func (d *RDSDatabase) rdsDatabaseProvision(installationID string, logger log.Fie
 		return err
 	}
 
-	encryptionKey, err := d.client.kmsCreateSymmetricKey(awsID, "Key used for encrypting RDS database")
+	keyAliasName := KMSAliasNameRDS(awsID)
+	keyMetadata, err := d.client.kmsGetSymmetricKey(keyAliasName)
 	if err != nil {
-		return errors.Wrapf(err, "unable to create RDS encryption key for installation %s", installationID)
-	}
+		if !IsErrorCode(err, kms.ErrCodeNotFoundException) {
+			return errors.Wrapf(err, "unable to verify encryption key %s for the RDS cluster ID %s", keyAliasName, awsID)
+		}
 
-	err = d.client.kmsCreateAlias(*encryptionKey.KeyId, KMSAliasNameRDS(awsID))
-	if err != nil && !IsErrorCode(err, kms.ErrCodeAlreadyExistsException) {
-		deletionKeyErr := d.client.kmsScheduleKeyDeletion(*encryptionKey.KeyId, KMSMinTimeEncryptionKeyDeletion)
-		if deletionKeyErr != nil {
-			logger.WithError(deletionKeyErr).Errorf("Failed to schedule encryption key %s for deletition", *encryptionKey.KeyId)
+		keyMetadata, err = d.client.kmsCreateSymmetricKey("Key used for encrypting RDS database")
```

Comment:
nit: it would be really nice to include the installation ID in the key description